src/components/Chat/Chat.tsx

```diff
@@ -93,14 +217,22 @@ const ChatWithLLM: React.FC<ChatWithLLMProps> = ({
   const [userTextFieldInput, setUserTextFieldInput] = useState<string>('')
   const [askText] = useState<AskOptions>(AskOptions.Ask)
   const [loadingResponse, setLoadingResponse] = useState<boolean>(false)
+  const [loadAnimation, setLoadAnimation] = useState<boolean>(false)
```

Comment:
it seems like we use loadingResponse and loadAnimation as basically the same thing - could you remove one of them please?

Reply:
They're not the same thing. `loadAnimation` is the state that triggers the LoadingDots when we are waiting for a response from streamingResponse. If you use a remote API or have a slow machine, then the latency is very apparent and there needs to exist some way to know that everything is working properly and the LLM is going to send a response. `loadAnimation` gets set to false on the very first receive from `streamingResponse`

`loadingResponse` continues until we are completely done from getting a response from the LLM. That is to say that `loadAnimation` stops on the first data received, while `loadingResponse` stops at the very end. It's purpose is to prevent the user from submitting another message or input while we are still streaming.

Reply:
ah gotcha. In that case to make the code easy to understand perhaps using a union type for all the states in loading response would be nice?